crates/proto/src/clients/mod.rs

```diff
@@ -0,0 +1,348 @@
+/// // Create a client with TLS and timeout
+/// let client: RpcClient = Builder::new()
+///     .with_address("https://rpc.example.com:8080".to_string())
+///     .with_tls()
+///     .with_timeout(Duration::from_secs(30))
+///     .connect::<Rpc>()
+///     .await?;
+/// # Ok(())
+/// # }
+/// ```
+#[derive(Default, Clone)]
+pub struct Builder {
+    /// Whether to enable TLS encryption for the connection.
+    pub with_tls: bool,
+    /// The gRPC server address in the format `{protocol}://{hostname}:{port}`.
+    pub address: String,
+    /// Optional timeout for gRPC operations.
+    pub with_timeout: Option<Duration>,
+    /// Optional version string to include in request metadata.
+    pub metadata_version: Option<String>,
+}
+
+impl Builder {
+    /// Creates a new builder with default configuration.
+    ///
+    /// The default configuration has TLS disabled, no timeout, and no metadata version.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Enables TLS encryption for the gRPC connection.
+    ///
+    /// When enabled, the client will use native system certificate roots for TLS verification.
+    /// This is required for connecting to secure endpoints.
+    #[must_use]
+    pub fn with_tls(mut self) -> Self {
+        self.with_tls = true;
+        self
+    }
+
+    /// Sets a timeout for gRPC operations.
+    ///
+    /// This timeout applies to all gRPC calls made through the client. If not set,
+    /// operations may hang indefinitely.
+    ///
+    /// # Arguments
+    ///
+    /// * `timeout` - The duration after which gRPC operations will timeout
+    #[must_use]
+    pub fn with_timeout(mut self, timeout: Duration) -> Self {
+        self.with_timeout = Some(timeout);
+        self
+    }
+
+    /// Sets the version string to include in request metadata.
+    ///
+    /// This version is used by the [`MetadataInterceptor`] to set the `Accept` header
+    /// in gRPC requests. If not set, the current crate version will be used.
+    ///
+    /// # Arguments
+    ///
+    /// * `version` - The version string to include in request metadata
+    #[must_use]
+    pub fn with_metadata_version(mut self, version: String) -> Self {
+        self.metadata_version = Some(version);
+        self
+    }
+
+    /// Sets the gRPC server address.
+    ///
+    /// The address should be in the format `{protocol}://{hostname}:{port}`.
+    /// Examples: `http://localhost:8080`, `https://api.example.com:443`
+    ///
+    /// # Arguments
+    ///
+    /// * `address` - The gRPC server address
+    #[must_use]
+    pub fn with_address(mut self, address: String) -> Self {
+        self.address = address;
+        self
+    }
+
+    /// Establishes an eager connection to the gRPC server.
+    ///
+    /// This method attempts to connect to the server immediately and returns a client
+    /// only after the connection is established. If the connection fails, an error
+    /// is returned.
```

Comment:
nit: here and in other places in this file it seems like we may be wrapping comments a bit too early (i.e., we usually do this around 100 chars.